src/lib/dnssd/IncrementalResolve.cpp

```diff
@@ -201,6 +184,7 @@ CHIP_ERROR IncrementalResolver::InitializeParsing(mdns::Minimal::SerializedQName
 
             Platform::CopyString(mSpecificResolutionData.Get<DnssdNodeData>().instanceName, nameCopy.Value());
         }
+        mSpecificResolutionData.Get<DnssdNodeData>().hasZeroTTL = (ttl == 0) ? false : true;
```

Comment:
`(condition) ? false : true` is an antipattern (and in this case it seems reversed. it should be just `condition`

Reply:
updated with proper check